### ISSUE TITLE
NullPointerException in TrayIcon on JDK 17/Linux Mint [with SSCCE]

### ISSUE THREAD

**Issue by stefan-reich** [JFormDesigner/FlatLaf#405]
Hi, here is a short code snippet that produces the exception. It's not fatal, in fact the tray icon works fine, but the exception is triggered quite a few times.

```
javac -cp flatlaf-1.6.1.jar TrayIconBugDemo.java 
java -cp flatlaf-1.6.1.jar:. TrayIconBugDemo
```

TrayIconBugDemo.java:

```
import java.awt.*;
import javax.imageio.*;
import java.net.*;
import java.util.*;
import java.io.*;

class TrayIconBugDemo {

  public static void main(String[] args) throws Exception {
    com.formdev.flatlaf.FlatLightLaf.setup();

    PopupMenu menu = new PopupMenu();
    menu.add("Test");

    Image image = ImageIO.read(new ByteArrayInputStream(Base64.getDecoder().decode(imageData)));
    TrayIcon trayIcon = new TrayIcon(image, "Right-click to produce exception on console", menu);
    SystemTray.getSystemTray().add(trayIcon);

    System.out.println("Right-click on gazelle tray icon to produce exception");
  }

  static String imageData =
    "iVBORw0KGgoAAAANSUhEUgAAAEAAAAAkCAYAAAA5DDySAAAABmJLR0QA/wD/AP+gvaeTAAAACXBIWXMA"
  + "AAsTAAALEwEAmpwYAAAAB3RJTUUH5QocEwcgsG17MQAABnZJREFUaN7lWX9Mk9cafvrx0a62mEpK+eHC"
  + "RSTObqtX8cJYZkl0ETROUaZZWJAZnXr14uXG+4f4h5u5yZ2yZUv0ekcuJm6hLtuicZtuTZMJWcwyXOPi"
  + "HxoQJ44Lemv0ovaHFvrr2R+eNqW2tJoiGJ7kpF9P3p7vPO8553nf8xYYPzwP4FUAEiYxMsZhzGcA/ABg"
  + "GMDnAIgpghkA/gkgBODPSWzVkiT9DcCWyb5DUoEE4C9ipQmgJon9agDDWq2WDoeDZrN5SKVSLXxayTcC"
  + "uBpF/pUxbDcC6A3b5uXl8f79+yTJjo4OLlu27EcAFU8D6QwACwH8T5AJAugCYEiwO0qjiYdbSUkJR0ZG"
  + "GI2jR4/SZDJ9kmCsSYEFAC5EEfkVwAsAFHFsSwFcEprA2LZgwQIGAgHGwuVy8dixY/cBfDqZiD8H4LMo"
  + "AhcAvJnAdj6A4/FIR7clS5ZwLNy7d48Gg8EF4G0AyokkrwHgjJr8NQBfAPgWwE8AFgk7ZUZGRmcy4uGW"
  + "m5vLGzduJHSAz+fjnDlzwvYjAF6bqPPeo1Qq45EYEqKnz83N/Xj//v2hO3fu0Ofz0WazcefOnSwpKYnY"
  + "azQabt68mTabje3t7ZH+pqYmhkKhhxzg9/tpMpkidgqFgqtXrz77pB3xsVKpdHd3d8eSvwRAp9fr/759"
  + "+/aEq+j1ellVVcX58+eP6q+trR01nizLtNvtoxwRDAZZVlYWsWlsbCRJtre3U6fT/QKgIIH2pAcKhWK3"
  + "RqP5esaMGSTJAwcOhFeiBcA7Bw8evDEwMJCQfGNjI6dPn04AVKvV9Hg8kZU1Go1xj0VVVRWHh4dJkqFQ"
  + "iGazmQCo1+sZDAYjY7vdblosloBarf4OgD4Jj+kAygG8/ij8y4uKijrXrl17RavV0uv1cmhoKKhWq0cA"
  + "cO/evXFJDw8P8/jx48zMzHyI3K5duyLiplarx9QHi8XCUCjE6upqZmRksL+/P6FONDQ0UKfTNQNYIhKy"
  + "fwH4HsAgAOp0OlZUVBDAypSYq1QqrV6v7718+fLd8IQGBwdJknv27LkOgKWlpQ9NpquriwUFBWMS83g8"
  + "PHPmTEoiWVRUxPLycjY1NTEZmpub446xfv16er1erlu3riPlnT9t2rTOoaGhSytXrmRfXx/7+vro9/sj"
  + "2lRQUODXaDSRGG6327lixYqUSLW2trK+vp6pRgoAdLvdSR1w69atUb9ZunQpz507R5I8ceJEsgx11Op/"
  + "feXKFafVauXNmzfjvqytrY1KpZIDAwOsq6tLmYhCoaDRaAzJshx6FAesWrWKqWDx4sWUZZmdnZ2RPo/H"
  + "Q4PBcHIsznLU8zs2m2357NmzlS0tLVi+fPkow+7ubly8eBFXr16Fz+dDcXGxPxAIOAA4RFocfv4vgFpx"
  + "6cG8efOwdetWtLS0OHt6empEUlMk0uk/ic+EsFqtcDgcyM/PH3PxNmzY4L5+/XrzoUOHXjhy5MhCvV5f"
  + "RlK6ffv2xuTXOUla1traSqfTyYaGBlqtVt++fftur1mz5prBYLgE4CSAd8VN7w9JhqvNyspiZWUl7XY7"
  + "XS4XZ86c2RPj7FjkA6gD8B+RZQ4CuCvuGSwvL0+6AyorK8/GCYtqMnk54iWDwUCj0cicnBzKsrwbQKEI"
  + "L+pHjJ5NNTU1kZDX29vLnJycrx4jlVUDyAEwS1SV9s+dO5dtbW1xyZ8/f54A3nrckP8ygDzhdSeArMcu"
  + "DEhSCQCqVKqw2B0Ni2saUpMvAVCSJG7ZsoWnT5+OXKc3bdoUTEfuYwTQkIZxTkaJ2Lw05ma5saIqyzLr"
  + "6+tpMpk+mEw3x4rw2RVJSTrxxhgRwyKcNOGQAfw/6sKU7vLbz0lC519FYXZCsUFMxisUPp0oTCF/+G2M"
  + "esXozG8cnXALQDaAKgAdaR47B8CLIkrMEnnFs0LAs7Va7SxZluHxeLoCgcAqsSOfOP4hVsMfUyaXAfwi"
  + "9GH2OBwR5OfnO1wuF/v7+5mXl/fhRB2DQvHnSHhbfiNEam9U31mxgmlFZmZm8alTp4IkuWjRIvdE68FG"
  + "IYbBmHO6dTxfmp2d/Q1JWiwWiprmhGMhgPdELbHwCbwva9u2beSDXLgRUxGFhYWbnE4n6+rqjmGKQlVd"
  + "XX338OHDNzGF8eqOHTs4HmL71MBsNlsB/Ft8/XQq+uB5SZJGABwSEej9qeiEj2LC8PvjnQpPNkgAygAU"
  + "A/gjHvx5e/Z3MOIGCJ7714gAAAAASUVORK5CYII=";
}
```

Oh and here's the stack trace:

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.awt.Toolkit.getDesktopProperty(String)" is null
	at java.desktop/sun.awt.X11.XToolkit.isRightMouseButton(XToolkit.java:2034)
	at java.desktop/sun.awt.X11.XBaseMenuWindow.doHandleJavaMouseEvent(XBaseMenuWindow.java:1031)
	at java.desktop/sun.awt.X11.XPopupMenuPeer.handleEvent(XPopupMenuPeer.java:292)
	at java.desktop/sun.awt.X11.XBaseMenuWindow$3.run(XBaseMenuWindow.java:949)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```

**Comment by DevCharly** [comment]
Can now reproduce it on Kubuntu...

**Comment by DevCharly** [comment]
This is actually an issue in JDK...

If `Toolkit.addPropertyChangeListener()` is invoked before `Toolkit.getDesktopProperty()` (which FlatLaf does),
then [XToolkit.initializeDesktopProperties()](https://github.com/openjdk/jdk/blob/6b55ef3b58ae405c46f440e06b5adb03cec3a500/src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java#L1507-L1527) is not invoked and desktop property `awt.mouse.numButtons` (and some others) are missing, which causes the NPE.

The problem is that `XToolkit.addPropertyChangeListener()` invokes `XToolkit.initXSettingsIfNeeded()`, which adds desktop properties to field `Map Toolkit.desktopProperties` [here](https://github.com/openjdk/jdk/blob/6b55ef3b58ae405c46f440e06b5adb03cec3a500/src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java#L1630) and [here](https://github.com/openjdk/jdk/blob/6b55ef3b58ae405c46f440e06b5adb03cec3a500/src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java#L1634).

Then the map `Toolkit.desktopProperties` is no longer empty, which prevents that `XToolkit.initializeDesktopProperties()` is invoked from `Toolkit.getDesktopProperty()` [here](https://github.com/openjdk/jdk/blob/6b55ef3b58ae405c46f440e06b5adb03cec3a500/src/java.desktop/share/classes/java/awt/Toolkit.java#L1544-L1546).

Windows is not affected by this issue because it calls `Toolkit.getDesktopProperty()` very early in `UIManager.<clinit>`  [here](https://github.com/openjdk/jdk/blob/6b55ef3b58ae405c46f440e06b5adb03cec3a500/src/java.desktop/share/classes/javax/swing/UIManager.java#L389).
macOS is not affected because FlatLaf does not invoke `Toolkit.addPropertyChangeListener()` on this platform.

There is a simple **workaround**. Add following before calling `FlatLightLaf.setup()`:

~~~java
Toolkit.getDefaultToolkit().getDesktopProperty( "dummy" );
~~~

I'll add this also to FlatLaf...

**Comment by DevCharly** [comment]
fixed in main branch. Thanks for reporting.

**Comment by DevCharly** [comment]
Fixed in [1.6.2](https://github.com/JFormDesigner/FlatLaf/releases/tag/1.6.2).
